libs/sdk-core/src/persist/sync.rs

```diff
@@ -162,17 +162,20 @@ impl SqliteStorage {
         // sync remote payments_external_info table
         tx.execute(
             "
-         INSERT into sync.payments_external_info
+         INSERT OR REPLACE INTO sync.payments_external_info
          SELECT
-          payment_id,
-          lnurl_success_action,
-          ln_address,
-          lnurl_metadata,
-          lnurl_withdraw_endpoint,
-          attempted_amount_msat,
-          attempted_error
+          remote_sync.payments_external_info.payment_id,
+          remote_sync.payments_external_info.lnurl_success_action,
+          remote_sync.payments_external_info.ln_address,
+          remote_sync.payments_external_info.lnurl_metadata,
+          remote_sync.payments_external_info.lnurl_withdraw_endpoint,
+          remote_sync.payments_external_info.attempted_amount_msat,
+          remote_sync.payments_external_info.attempted_error,
+          remote_sync.payments_external_info.external_metadata,
+          remote_sync.payments_external_info.updated_at
          FROM remote_sync.payments_external_info
-         WHERE payment_id NOT IN (SELECT payment_id FROM sync.payments_external_info);",
+         LEFT JOIN sync.payments_external_info ON sync.payments_external_info.payment_id = remote_sync.payments_external_info.payment_id
+         WHERE remote_sync.payments_external_info.updated_at > sync.payments_external_info.updated_at;",
```

Comment:
The problem with this approach is that if one device updates the metadata and wasn't synced other devices will override their data on next sync (since they would update the whole item)
Perhaps it is better to use a different table for metadata?